WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsReportCard.swift

```diff
@@ -50,14 +50,18 @@ struct AnalyticsReportCard: View {
                         .shimmering(active: isRedacted)
 
                     AdaptiveStack(horizontalAlignment: .leading) {
-                        DeltaTag(value: leadingDelta, backgroundColor: leadingDeltaColor, textColor: leadingDeltaTextColor)
-                            .frame(maxWidth: .infinity, alignment: .leading)
-                            .redacted(reason: isRedacted ? .placeholder : [])
-                            .shimmering(active: isRedacted)
-
-                        AnalyticsLineChart(dataPoints: leadingChartData, lineChartColor: leadingChartColor)
-                            .aspectRatio(Layout.chartAspectRatio, contentMode: .fit)
-                            .frame(maxWidth: scaledChartWidth)
+                        if let leadingDelta, let leadingDeltaColor, let leadingDeltaTextColor {
```

Comment:
It doesn't seem correct that we have 3-4 separate vars instead of a single DeltaPercentage struct object.
With added optionality it's not only code duplication, but possible intermediate states, where _some_ of those properties belonging to a single object can be `nil`. These states are unreachable in runtime, but we have to manually handle them in UI logic 🤔

We discussed earlier that it's preferred approach to prevent injecting any logic into views, so it's not a blocker for this PR, but a highlight of a counter-example for that theory.

Reply:
I agree this feels a little odd but it does offer the most flexibility for the view. If we wanted to spend more time on this, we could probably improve it by separating these optional properties from the ability to render the view they belong to.

@Ecarrion just want to loop you in here so you're aware of the discussion.